sql/handler.cc

```diff
@@ -7369,8 +7376,9 @@ bool Vers_parse_info::fix_alter_info(THD *thd, Alter_info *alter_info,
           return true;
         }
 
-        if (vers_change_sys_field(thd, name, alter_info,
-                                  f->flags & VERS_SYSTEM_FIELD, name))
+        bool integer= table->vers_start_field()->type() == MYSQL_TYPE_LONGLONG;
+        if (vers_change_sys_field(thd, name, alter_info, f->flags & VERS_SYSTEM_FIELD, name,
+                                  integer))
```

Comment:
Remove `vers_change_sys_field()`.

Reply:
Are you sure? That will change UI. `DROP sys_start, DROP sys_end` is currently required step before `DROP SYSTEM VERSIONING` as requested by @vuvova. I'm gonna ask him about this anyways but this PR is only about buggy versioning switch.

Reply:
I mean unwrap code from `vers_change_sys_field()`. The function is redundant, too much args, some duplicated.

Reply:
Done.